include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -18,13 +18,14 @@ def CircuitOp : FIRRTLOp<"circuit",
     The "firrtl.circuit" operation represents an overall Verilog circuit,
     containing a list of modules.
   }];
-  let arguments = (ins StrAttr:$name);
+  let arguments = (ins StrAttr:$name, OptionalAttr<ArrayAttr>:$annotations);
   let results = (outs);
   let regions = (region SizedRegion<1>:$body);
 
   let skipDefaultBuilders = 1;
   let builders = [
-    OpBuilder<(ins "StringAttr":$name)>
+    OpBuilder<(ins "StringAttr":$name, CArg<"ArrayRef<DictionaryAttr>",
+               "ArrayRef<DictionaryAttr>()">:$annotations)>
```

Comment:
This can take an ArrayAttr instead of ArrayRef<dict>

Reply:
Thanks. This is cleaner. I applied this suggestion in a later `!squash` commit.